File: go/buscabinaria/buscabinaria.go
package main

import (
	"fmt"
	"strconv"
)

func busca_binaria(list []int, valor int) int {
	menor := 0;
	maior := len(list) -1;

	for menor <= maior {
		meio := (menor + maior) / 2;
		valorMeio := list[meio];

		if valor == valorMeio {
			return valor;
		}

		if(valor > valorMeio) {
			menor = meio + 1;
		}

		if(valor < list[meio]) {
			maior = meio - 1;
		}

	}
	return -1;	
}

func main() {
	list := []int{1,5,8,12,20,45};

	result := busca_binaria(list,20);

	if(result == -1) {
		fmt.Println("Numero não encontrado")
	} else {
		fmt.Println(strconv.Itoa(result) + " foi encontrado");
	}

}

File: buscaBinaria/buscabinaria.py
def busca_binaria(lista, item):
  menor = 0;
  maior = len(lista) - 1;

  while(menor <= maior):
    meio = (menor + maior) // 2
    valorMeio = lista[meio]

    if valorMeio == item:
      return item
    if valorMeio > item:
      maior = meio - 1
    else:
      menor = meio + 1
  
  return None

lista = [1, 5, 7,9]

print(busca_binaria(lista, 3))
print(busca_binaria(lista, 9))